docs/integrations/destinations/s3-migrations.md

```diff
@@ -0,0 +1,202 @@
+
+| Old field name          | New field name           | Json Type          | Avro Type                                                 | Parquet Type                                 | Description                                                                                            |
+|-------------------------|--------------------------|--------------------|-----------------------------------------------------------|----------------------------------------------|--------------------------------------------------------------------------------------------------------|
+| `_airbyte_ab_id`        | `_airbyte_raw_id`        | string (UUID)      | `{ "type": "string", "logicalType": "uuid" }`             | `String`                                     | Airbyte-added unique identifier.                                                                       |
+| `_airbyte_emitted_at`   | `_airbyte_extracted_at`  | integer (epoch ms) | `{ "type": "string", "logicalType": "timestamp-millis" }` | `Int64(logicalType=Timestamp(Milliseconds))` | Time at which the data was loaded from the source.                                                     |
+| [NEW FIELD]             | `_airbyte_generation_id` | integer            | `{ "type": "long" }`                                      | `Int64`                                      | Monotonically-increasing refresh id, if applicable.                                                    |
+| [NEW FIELD]             | `_airbyte_meta`          | object (see below) | Record (see below)                                        | Record (see below)                           | Additional metadata, including change data capture info and sync id.                                   |
+| `_airbyte_data`         | [UNCHANGED]              | optional object    | (from client schema)                                      | (from client schema)                         | Data payload when [flattening is disabled](https://docs.airbyte.com/integrations/destinations/s3#csv). |
+
+The `_airbyte_meta` field is an object that currently has one field:
+
+| Field Name     | Json Type | Description                                                    |
+|----------------|-----------|----------------------------------------------------------------|
+| `changes`      | list      | Change data capture info                                       |
+| `sync_id`      | integer   | Monotonically-increasing integer representing the current sync |
+
+The `changes` field is a list of objects, each of which represents a change to the data. Each object has the following fields:
+
+| Field Name | Json Type* | Description                                                                |
+|------------|------------|----------------------------------------------------------------------------|
+| `field`    | string     | The name of the field that was affected                                    |
```

Comment:
this is the jsonpath, right? or do we actually only capture this for top-level fields?

Reply:
I can't remember. The path is passed into the listener, but I think I just use the field name. It's an easy fix though, I'll check.

Reply:
Specifically this is how it behaves right now:
* any field failure at any depth triggers a capture
* however, only the field name is used

The reason for this was because it was a choice between the original name (which might be ambiguous if there nested objects with the same fields) and a path made of avro-mangled names.

The path is more informative though, and technically matches what they'd be looking at in their data lake, so it's probably the better choice?

Reply:
Actually, I'm going to skip on this and pocket it as an improvement. The path passed out of the json-to-avro converter is not usable, because it doesn't account for array indexes. Fixing it is a major change.

Reply:
https://app.zenhub.com/workspaces/destinations-6332532d6893ce001c7a56a7/issues/gh/airbytehq/airbyte/43397